mteb/tasks/Retrieval/multilingual/XQuADRetrieval.py

```diff
@@ -0,0 +1,112 @@
+        eval_langs=_LANGUAGES,
+        main_score="ndcg_at_10",
+        date=("2019-05-21", "2019-11-21"),
+        form=["written"],
+        domains=["Web"],
+        task_subtypes=[],
+        license="CC BY-SA 4.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""@article{Artetxe:etal:2019,
+      author    = {Mikel Artetxe and Sebastian Ruder and Dani Yogatama},
+      title     = {On the cross-lingual transferability of monolingual representations},
+      journal   = {CoRR},
+      volume    = {abs/1910.11856},
+      year      = {2019},
+      archivePrefix = {arXiv},
+      eprint    = {1910.11856}
+}
+@inproceedings{
+      dumitrescu2021liro,
+      title={LiRo: Benchmark and leaderboard for Romanian language tasks},
+      author={Stefan Daniel Dumitrescu and Petru Rebeja and Beata Lorincz and Mihaela Gaman and Andrei Avram and Mihai Ilie and Andrei Pruteanu and Adriana Stan and Lorena Rosia and Cristina Iacobescu and Luciana Morogan and George Dima and Gabriel Marchidan and Traian Rebedea and Madalina Chitez and Dani Yogatama and Sebastian Ruder and Radu Tudor Ionescu and Razvan Pascanu and Viorica Patraucean},
+      booktitle={Thirty-fifth Conference on Neural Information Processing Systems Datasets and Benchmarks Track (Round 1)},
+      year={2021},
+      url={https://openreview.net/forum?id=JH61CD7afTv}
+}""",
+        n_samples={"test": 1190},
+        avg_character_length={"test": 788.7},
+    )
+
+    def load_data(self, **kwargs):
+        if self.data_loaded:
+            return
+
+        split = "validation"
+        queries = {lang: {split: {}} for lang in self.langs}
+        corpus = {lang: {split: {}} for lang in self.langs}
+        relevant_docs = {lang: {split: {}} for lang in self.langs}
+
+        for lang in self.langs:
+            data = datasets.load_dataset(
+                name=f"xquad.{lang}", **self.metadata_dict["dataset"]
+            )[split]
+            data = data.filter(lambda x: x["answers"]["text"] != "")
+
+            question_ids = {
+                question: sha256(question.encode("utf-8")).hexdigest()
+                for question in set(data["question"])
+            }
+            context_ids = {
+                context: sha256(context.encode("utf-8")).hexdigest()
+                for context in set(data["context"])
+            }
+
+            for row in data:
+                question = row["question"]
+                context = row["context"]
+                query_id = question_ids[question]
+                queries[lang][split][query_id] = question
+
+                doc_id = context_ids[context]
+                corpus[lang][split][doc_id] = {"text": context}
+                if query_id not in relevant_docs[lang][split]:
+                    relevant_docs[lang][split][query_id] = {}
+                relevant_docs[lang][split][query_id][doc_id] = 1
```

Comment:
I think you can simplify this by using the `id` provided in the dataset, and build an id for queries by enumerating them (this dataset exists for long enough I'm not sure it contains duplicates). See example [here](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/fra/FQuADRetrieval.py) and [here](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/tasks/Retrieval/fra/SyntecRetrieval.py)

Reply:
I have made the necessary revisions as suggested in your comments above.